compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java

```diff
@@ -263,14 +261,18 @@ public boolean isAnydata() {
         if (this.isAnyData.isPresent()) {
             return this.isAnyData.get();
         }
-
+        if (this.resolvingAnyData) {
+            return false;
```

Comment:
Shouldn't we be returning `true` if we are resolving?

Can you check if `type Foo int|map<Foo>` correctly resolves to `anydata`?

Reply:
Yes, it should be true. Fixed.